src/security/ldap/src/main/java/org/geoserver/security/ldap/BindingLdapAuthoritiesPopulator.java

```diff
@@ -209,10 +209,12 @@ public Set<GrantedAuthority> getGroupMembershipRoles(
                     // Get ldap groups in form of Pair<String,String> -> Pair<name,dn>
                     final String formattedFilter =
                             MessageFormat.format(groupSearchFilter, userDn, username);
+                    // Replace the escape token "\" with "\\" to search.
```

Comment:
this is superfluous as the method is documented, but why could be documented

```suggestion
                    // search requires an escaped string
```

Reply:
Thanks for your review. Great.
I'm not a real Java developer and I'm not exactly sure what you mean.

Is your suggestion to replace
   `String formattedFilterNew = LDAPUtils.escapeSearchString(formattedFilter);`
with 
   `formattedFilter = LDAPUtils.escapeSearchString(formattedFilter);`?

Or/and replace all calls to `escapeSearchString` with `searchString.replace("\\", "\\\\")`"?

I will follow your suggestion for the text change.

Reply:
And I will follow your suggestion for the comment change.

Reply:
Yes, my suggestion is to use `formattedFilter = LDAPUtils.escapeSearchString(formattedFilter);`

Reply:
Fine, I shall implement your suggestion one of these days.